### PR TITLE
Fixes TypeError when using astropy.io.votable.validate( ... , xmllint=True) - #7255

### LINKED ISSUE #7255: TypeError when using astropy.io.votable.validate( ... , xmllint=True)
When xmllint = True, 
validation goes through **astropy.io.votable.table.py** line 306 
```
 if xmllint and os.path.exists(filename):
        from ...utils.xml import validate

        if votable is None:
            version = "1.1"
        else:
            version = votable.version
        success, stdout, stderr = validate.validate_schema(
            filename, version)
```
Where **astropy.utils.xml.validate.py** validate_schema is called with schema_file as the votable version - like "1.1" for example
`validate_schema(filename, schema_file):`
```
    base, ext = os.path.splitext(schema_file)
    if ext == '.xsd':
        schema_part = '--schema ' + schema_file
    elif ext == '.dtd':
        schema_part = '--dtdvalid ' + schema_file
    else:
        raise TypeError("schema_file must be a path to an XML Schema or DTD")
```
Thing being that xml.validate.py validate() is expecting a ".xsd" or a ".dtd" and throws a TypeError exception line 42 otherwise. 

table.py should be calling **astropy.io.votable.xmlutil.py** validate_schema() instead. 
`validate_schema(filename, version='1.1'): ...`

### PR DESCRIPTION
**Pull request by mmebsout**
@pllim I fixed the bug and added one unit test in table_test.py

EDIT: Fix #7255

### CODE REVIEW

**Comment by pllim** [comment]
Please add a change log under "bug fix" for 3.0.1. I will do a thorough review next week. Thanks!

**Comment by pllim** [comment]
This has slipped to 3.0.2. I was going to move the change log for you, but that section has not been added in `master` yet. I'll come back later this week to rebase and move the change log, unless you get to it first. After that, I'll review. Thank you for your patience!

**Comment by mmebsout** [comment]
@pllim No worries :) I'll try to do it tonight .

**Comment by mmebsout** [comment]
@pllim Sorry I did a merge instead of rebase. I'll fix it today.

**Comment by mmebsout** [comment]
Re-wrote the commits in a clean branch 
Fixes #7255

**Comment by pllim** [comment]
Would you mind squashing it, or me doing it for you? Just wondering.

**Comment by mmebsout** [comment]
@pllim I would appreciate if you do it. Thank you :)

**Comment by pllim** [comment]
Okay, no problem. I'll take care of this today.

File: astropy/io/votable/tests/table_test.py [hunk 1]
```diff
@@ -177,3 +177,16 @@ def test_empty_table():
...
+
+def test_validate_xmllint_true(monkeypatch):
+    from ..table import validate
+    from .. import xmlutil
+
+    file = get_pkg_data_filename('data/empty_table.xml')
+
+    def mock_xmllint_validate(filename, version):
+        return 0, 'ok', 'ko'
+    monkeypatch.setattr(xmlutil, 'validate_schema', mock_xmllint_validate)
```
Comment (pllim) [inline astropy/io/votable/tests/table_test.py]: Hmm. I wonder if it is possible to only mock the actual `xmllint` subprocess. I'll investigate.

Comment (mmebsout) [reply astropy/io/votable/tests/table_test.py]: I think you'd have to be able to import it to do so.  The logic in the xmlutil.validate_schema is already tested in vo_test with a try except that's why I mocked it direclty there.

Comment (pllim) [reply astropy/io/votable/tests/table_test.py]: Ah, ops. I saw this too late! :sweat_smile:

Comment (mmebsout) [reply astropy/io/votable/tests/table_test.py]: No worries :) whatever fits best the astopy way to do things ^^ is the way to go and you should know that better than me ;)

**Comment by pllim** [comment]
Thank you for your understanding and patience, @mmebsout ! Test failure is unrelated (xref #7298). Merging.

**Comment by mmebsout** [comment]
@pllim All the same :) have a great week-end !

**Pull request merged**